build fails if sqlite is missing

Building ACT fails when the `sqlite` library is missing, due to an unused variable 'msg_handler'. Full make.err: 
```
/home/simho/software/act/ACT/src/act/molprop/molprop_sqlite3.cpp: In function 'void alexandria::ReadSqlite3(MsgHandler*, const char*, std::vector<MolProp>*, double)':
/home/simho/software/act/ACT/src/act/molprop/molprop_sqlite3.cpp:231:51: error: unused parameter 'msg_handler' [-Werror=unused-parameter]
  231 | void ReadSqlite3(MsgHandler                      *msg_handler,
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/actLib.dir/build.make:678: src/CMakeFiles/actLib.dir/act/molprop/molprop_sqlite3.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:2722: src/CMakeFiles/actLib.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:166: all] Error 2

```
Since `sqlite` is considered an optional dependency, maybe it would be better to raise a warning and continue the compilation?

--- a/src/act/molprop/molprop_sqlite3.cpp
+++ b/src/act/molprop/molprop_sqlite3.cpp
@@ -228,7 +228,7 @@ static void getClasses(MsgHandler           *msg_handler,
 }
 #endif
 
-void ReadSqlite3(MsgHandler                      *msg_handler,
+void ReadSqlite3(gmx_unused MsgHandler           *msg_handler,
                  gmx_unused const char           *sqlite_file,
                  gmx_unused std::vector<MolProp> *mp,
                  gmx_unused double                ref_temperature)

